Create Documentation for the ComInterfaceGenerator (#36833)
dotnet/docs: docs/standard/native-interop/customize-struct-marshalling.md

Sometimes the default marshalling rules for structures aren't exactly what you need. The .NET runtimes provide a few extension points for you to customize your structure's layout and how fields are marshalled. Customizing structure layout is supported for all scenarios, but customizing field marshalling is only supported for scenarios where runtime marshalling is enabled. If [runtime marshalling is disabled](disabled-marshalling.md), then any field marshalling must be done manually.

> [!NOTE]

> > This article does not cover customizing marshalling for source-generated interop. If using [source-generated interop for P/Invokes](./pinvoke-source-generation.md) or [COM](./comwrappers-source-generation.md), customizing marshalling is discussed [here](./custom-marshalling-source-generation.md).

Reviewer: ```suggestion
> This article doesn't cover customizing marshalling for source-generated interop. If you're using [source-generated interop for P/Invokes](./pinvoke-source-generation.md) or [COM](./comwrappers-source-generation.md), see [customizing marshalling](./custom-marshalling-source-generation.md).
```

> > This article does not cover customizing marshalling for source-generated interop. If using [source-generated interop for P/Invokes](./pinvoke-source-generation.md) or [COM](./comwrappers-source-generation.md), customizing marshalling is covered in [another article](./custom-marshalling-source-generation.md).

Reviewer 2: ```suggestion
> This article does not cover customizing marshalling for source-generated interop. If using [source-generated interop for P/Invokes](./pinvoke-source-generation.md) or [COM](./comwrappers-source-generation.md), customizing marshalling is discussed [here](./custom-marshalling-source-generation.md).
```

> ## Customizing structure layout

Reviewer: ```suggestion
## Customize structure layout
```
